### PR TITLE
feat: add libp2p http blog

### PR DESCRIPTION
**Pull request by p-shahi**
Delivering https://pl-strflt.notion.site/HTTP-P2P-HTTP-with-more-Ps-0e90fd9139964d7888be14e331e12815?pvs=4

### CODE REVIEW

**Review by thomaseizinger — APPROVED** [review]
Great stuff! Don't have anything to add :)

**Review by dhuseby — APPROVED** [review]
I reviewed it a it looks good to me. I made some comments on the notion page but I have no requests for content changes.

My only request for next time would be to have some more graphics illustrating streams vs send/response at the transport level, http request/response encoding happens at the protocol level and how the wire format differs in different combinations of HTTP transport and protocols.

File: src/_blog/http.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+<aside>
+💡 For more details about valid multiaddrs for the HTTP transport, refer to the [spec](https://github.com/libp2p/specs/blob/master/http/transport-component.md).
+
+</aside>
+
+The HTTP transport is a normal HTTPS server/client. There’s is nothing libp2p specific about it. This is on purpose as it allows us to interoperate with the wide existing HTTP ecosystem.
+
+## HTTP Semantics
+
+HTTP semantics differ from the HTTP transport in that they are the abstract form of HTTP. They don’t specify ***how*** an HTTP message will be encoded and sent to a remote node, they only specify ****what**** an HTTP message is and its interpretation. These semantics are defined by [RFC 9110](https://www.rfc-editor.org/rfc/rfc9110.html). This difference means we can adopt HTTP semantics without limiting ourselves to ****only**** an HTTP transport. We can use WebTransport, WebRTC, or a hole-punched QUIC connection to make an HTTP request. This allows developers to create applications using familiar HTTP tools in a p2p setting. On a technical level, this is implemented by opening a new stream for every HTTP request and encoding the HTTP message as HTTP/1.1.
```
Comment (mxinden) [inline src/_blog/http.md:59]:
```suggestion
HTTP semantics differ from the HTTP transport in that they are the abstract form of HTTP. They don’t specify **how** an HTTP message will be encoded and sent to a remote node, they only specify **what** an HTTP message is and its interpretation. These semantics are defined by [RFC 9110](https://www.rfc-editor.org/rfc/rfc9110.html). This difference means we can adopt HTTP semantics without limiting ourselves to **only** an HTTP transport. We can use WebTransport, WebRTC, or a hole-punched QUIC connection to make an HTTP request. This allows developers to create applications using familiar HTTP tools in a p2p setting. On a technical level, this is implemented by opening a new stream for every HTTP request and encoding the HTTP message as HTTP/1.1.
```

I don't think the blog post  needs to be so bold :P. A typo?

**Review by mxinden — APPROVED** [review]
Very well done!

**Pull request merged**